src/node_http2.cc

```diff
@@ -55,10 +55,27 @@ enum Http2OptionsIndex {
   IDX_OPTIONS_FLAGS
 };
 
+static uint32_t http2_padding_buffer[3];
+static uint32_t http2_options_buffer[IDX_OPTIONS_FLAGS + 1];
+static uint32_t http2_settings_buffer[IDX_SETTINGS_COUNT + 1];
+static double http2_session_state_buffer[IDX_SESSION_STATE_COUNT];
+static double http2_stream_state_buffer[IDX_STREAM_STATE_COUNT];
+
+static const size_t http2_options_buffer_byte_length =
+    sizeof(http2_options_buffer) * (IDX_OPTIONS_FLAGS + 1);
+static const size_t http2_settings_buffer_byte_length =
+    sizeof(http2_settings_buffer) * (IDX_SETTINGS_COUNT + 1);
+static const size_t http2_padding_buffer_byte_length =
+    sizeof(http2_padding_buffer) * 3;
+static const size_t http2_stream_state_buffer_byte_length =
+    sizeof(http2_stream_state_buffer) * IDX_STREAM_STATE_COUNT;
+static const size_t http2_session_state_buffer_byte_length =
+    sizeof(http2_session_state_buffer) * IDX_SESSION_STATE_COUNT;
```

Comment:
Yeah, initially that's what I had done but figured it was just as easy not to throw things into the environment. These are fixed sized things that are pretty specific to the http2 implementation. There's not much that embedders would be able to do to change them. The one that we will definitely have to look at later on are the settings -- right now we support only a fixed range of settings as determined by the underlying nghttp2 library, eventually we may want to open that up for extensions, in which case, the current implementation would need to be reworked entirely.